converter.go

```diff
@@ -536,55 +563,142 @@ func Array(a interface{}) interface{} {
 	var t snowflakeType
-	var arr []string
+	var arr []*string
 	switch reflect.TypeOf(nv.Value) {
 	case reflect.TypeOf(&intArray{}):
 		t = fixedType
 		a := nv.Value.(*intArray)
 		for _, x := range *a {
-			arr = append(arr, strconv.Itoa(x))
+			v := strconv.Itoa(x)
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&int64Array{}):
 		t = fixedType
 		a := nv.Value.(*int64Array)
 		for _, x := range *a {
-			arr = append(arr, strconv.Itoa(int(x)))
+			v := strconv.FormatInt(x, 10)
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&int32Array{}):
 		t = fixedType
 		a := nv.Value.(*int32Array)
 		for _, x := range *a {
-			arr = append(arr, strconv.Itoa(int(x)))
+			v := strconv.Itoa(int(x))
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&float64Array{}):
 		t = realType
 		a := nv.Value.(*float64Array)
 		for _, x := range *a {
-			arr = append(arr, fmt.Sprintf("%g", x))
+			v := fmt.Sprintf("%g", x)
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&float32Array{}):
 		t = realType
 		a := nv.Value.(*float32Array)
 		for _, x := range *a {
-			arr = append(arr, fmt.Sprintf("%g", x))
+			v := fmt.Sprintf("%g", x)
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&boolArray{}):
 		t = booleanType
 		a := nv.Value.(*boolArray)
 		for _, x := range *a {
-			arr = append(arr, strconv.FormatBool(x))
+			v := strconv.FormatBool(x)
+			arr = append(arr, &v)
 		}
 	case reflect.TypeOf(&stringArray{}):
 		t = textType
 		a := nv.Value.(*stringArray)
-		arr = *a
+		for _, x := range *a {
+			v := x // necessary for address to be not overwritten
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&byteArray{}):
+		t = binaryType
+		a := nv.Value.(*byteArray)
+		for _, x := range *a {
+			v := hex.EncodeToString(x)
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&timestampNtzArray{}):
+		t = timestampNtzType
+		a := nv.Value.(*timestampNtzArray)
+		for _, x := range *a {
+			v := strconv.FormatInt(x.UnixNano(), 10)
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&timestampLtzArray{}):
+		t = timestampLtzType
+		a := nv.Value.(*timestampLtzArray)
+		for _, x := range *a {
+			v := strconv.FormatInt(x.UnixNano(), 10)
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&timestampTzArray{}):
+		t = timestampTzType
+		a := nv.Value.(*timestampTzArray)
+		for _, x := range *a {
+			var v string
+			if stream {
+				v = x.Format(format)
+			} else {
+				_, offset := x.Zone()
+				v = fmt.Sprintf("%v %v", x.UnixNano(), offset/60+1440)
+			}
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&dateArray{}):
+		t = dateType
+		a := nv.Value.(*dateArray)
+		for _, x := range *a {
+			_, offset := x.Zone()
+			x = x.Add(time.Second * time.Duration(offset))
+			v := fmt.Sprintf("%d", x.Unix()*1000)
+			arr = append(arr, &v)
+		}
+	case reflect.TypeOf(&timeArray{}):
+		t = timeType
+		a := nv.Value.(*timeArray)
+		for _, x := range *a {
+			var v string
+			if stream {
```

Comment:
It's because how GS parses time when it's passed from a stream (aka via a file) and when it's passed directly as an array is different, so the conversions are different.